pkg/target/clusters.go

```diff
@@ -72,10 +63,39 @@ func NewClusters(mainCfg conf.Main, cfg conf.Clusters, lg *zap.Logger, ms *metri
 			return cls, fmt.Errorf("incorrect cluster type %s for cluster %s",
 				cl.Type, cl.Name)
 		}
-		cl.Type = cc.Type
 
 		cls[cl.Name] = &cl
 	}
 
-	return cls, nil
+	for _, cl := range cls {
+		go cl.keepAvailableHostsUpdated()
+		if cl.Type == conf.LB {
+			for _, h := range cl.Hosts {
+				h.Connect(cl.UpdateHostHealthStatus, 1)
```

Comment:
Why would we want different initial connection logic for *LB*, *toall*, and *jump*?

Reply:
This is so that the available hosts are updated from the start.